turbonomic/kubeturbo
# [TRB-41904] upgrade dependency k8s.io/kubernetes

# Intent

Fix scanner reported issues CVE-2023-2431, CVE-2023-2728, CVE-2023-2727
Caused by indirect dependencies from https://github.com/argoproj

# Background

Fix issues [TRB-41904](https://jsw.ibm.com/browse/TRB-41904), [TRB-42016](https://jsw.ibm.com/browse/TRB-42016), [TRB-42017](https://jsw.ibm.com/browse/TRB-42017)

Scanner report here: 
https://hyc-turbo-devops-team-jenkins.swg-devops.com/job/kubeturbo-whitesource-scanner/WhiteSource_20HTML_20Report/
The report recommends upgrading k8s.io/Kubernetes to v1.25.11
Trying higher upgrade (v1.27.3) breaks pkg dependencies from https://github.com/argoproj

# Testing

Check updated whitesource scanner report.
TwistLock scan [twistlock-scan-results-20230705-162317-565579000-UTC-06160807.results.csv](https://github.com/turbonomic/kubeturbo/files/11960159/twistlock-scan-results-20230705-162317-565579000-UTC-06160807.results.csv)

# Manual testing

- build image  using the changes

```bash
docker buildx build --platform=linux/amd64 --tag docker-na.artifactory.swg-devops.com/hyc-turbo-internal-team-docker-local/turbonomic/tamer/badkubeturbo:8.9.5-TM -f build/Dockerfile.multi-archs --build-arg VERSION=8.9.5-TM --push .
```
- deploy it to ROSA cluster using helm, test and check pod log for errors.
```yaml
image:
  #repository: icr.io/cpopen/turbonomic/kubeturbo
  repository:  docker-na.artifactory.swg-devops.com/hyc-turbo-internal-team-docker-local/turbonomic/tamer/badkubeturbo
  tag: 8.9.5-TM
  pullPolicy: Always
```

# Checklist

These are the items that must be done by the developer and by reviewers before the change is ready to merge. Please ~~strikeout~~ any items that are not applicable, but don't delete them

- [ ] Developer Checks
    - [X] Full build with unit tests and fmt and vet checks
    - [X] No unlicensed images, no third-party code (such as from StackOverflow)
    - [X] Manual testing done (and described)
    - [ ] Product sweep run and passed
    - [ ] ~~Developer wiki updated (and linked to this description)~~
- [ ] Reviewer Checks
    - [ ] Merge request description clear and understandable
    - [ ] Developer checklist items complete
    - [ ] Functional code review (how is the code written)
    - [ ] Architectural review (does the code try to do the right thing, in the right way)
    - [ ] Defensive coding (incoming data checked / sanitized, exceptions logged, clear error messages)
    - [ ] No unlicensed images, no third-party code (such as from StackOverflow)
    - [ ] Security review checklist complete.

# Audience

@ading1977 @irfanurrehman
